lib/filesystem.js

```diff
@@ -112,6 +112,8 @@ FileSystem.prototype.getItem = function(filepath) {
     if (item) {
       if (item instanceof Directory && name !== currentParts[i]) {
```

Comment:
The existing behaviour below might be of concern, later. Is there a better way to accommodate Windows?

See notes on why and my workaround: [here](https://github.com/tschaub/mock-fs/pull/304/files#diff-6d186b954a58d5bb740f73d84fe39073R260).